cloudwatch_logs_subscription/main.tf

```diff
@@ -1,4 +1,11 @@
+locals {
+  iam_role_arn  = var.iam_role_arn != "" ? var.iam_role_arn : aws_iam_role.this[0].arn
```

Comment:
Where is the 'arn' of aws_iam_role.this[0].arn defined? I'm guessing it's similar to the oid of resources we pass around, but I'm not seeing the definition in the resource below

Reply:
It is a computed field on aws_iam_role, so it is only populated after the resource has been created. See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#attributes-reference